### PR TITLE
Use python3 -m pip instead of pip

### PR DESCRIPTION
**Pull request by mppf**
Follow-up to PR #16560 (and the much older PR #9102)

* Update chpl-venv to check for Python 3.5 or newer
* Invoke pip with `python3 -m pip` to avoid issues with mixing Python and
  Pip versions. Updated documentation recommending that pip be run as
  well.
* Updates the GETPIP functionality to only run when `python3 -m pip` fails

- [x] verified expected error on a system with python 3.4
- [x] verified that `pip` is still downloaded on a system with python3 but no pip (typical for XCs)
- [x] full local testing after a `make clobber`

Reviewed by @lydia-duncan - thanks!

### CODE REVIEW

File: third-party/chpl-venv/Makefile [hunk 1]
```diff
@@ -32,22 +54,10 @@ check-exes: check-python
...
-	  $(GET_PIP_COMMAND) ) \
-	fi
-
 # Install virtualenv program.
-$(CHPL_VENV_VIRTUALENV): check-exes $(PIP)
-	export PYTHONPATH="$$CHPL_PYTHONPATH:$(PIPLIBS):$$PYTHONPATH" && \
-	$(PYTHON) $(PIP) install -U --force-reinstall --ignore-installed $(LOCAL_PIP_FLAGS) \
-	 --prefix=$(CHPL_VENV_INSTALL_DIR) $(CHPL_PIP_INSTALL_PARAMS) $(shell cat virtualenv.txt)
+$(CHPL_VENV_VIRTUALENV): check-exes
+	$(PIP) install -U --force-reinstall --ignore-installed $(LOCAL_PIP_FLAGS) \
```
Comment (lydia-duncan) [inline third-party/chpl-venv/Makefile:59]: You've removed the PIP definition but you still use it here, is that okay?  Or are you getting it from another Makefile?

Comment (lydia-duncan) [reply third-party/chpl-venv/Makefile:59]: Nevermind, I see it in the Makefile.include file

**Review by lydia-duncan — APPROVED** [review]
I think this looks good!

**Comment by ronawho** [comment]
pip is not always available with python3. e.g. it's not available with the system python3 on crays. Unless it's causing some problems, I'd prefer to keep the GETPIP logic.

**Pull request merged**